section 4_Flask_Restfull/app.py

```diff
@@ -0,0 +1,59 @@
+from flask import Flask,request
+from flask_restful import Resource, Api,reqparse
+from flask_jwt import JWT,jwt_required
+from security import authenticate, identity
+
+app = Flask(__name__)
+app.secret_key = 'jose'
+api = Api(app)
+
+jwt = JWT(app,authenticate,identity) # /auth end point dega
+
+items = []
+
+
+class Item(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument('price', type=float,required=True, help='this field cannot be left blank')
+
+    @jwt_required()
+    def get(self,name):
+        item = next(filter(lambda x:x['name']==name,items),None)
+        return item, 200 if item else 404
+
+    def post(self,name):
+        if next(filter(lambda x:x['name']==name,items),None) is not None:
+            return {"message":"name {} already exist".format(name)}, 400
```

Comment:
Ideally we should create a function that takes dictionary value as an arg I guess. We are creating this response dictionary every where which is not a very good idea because we are adding same key "message" in every response.